src/JsonSchema/Constraints/CollectionConstraint.php

```diff
@@ -111,4 +130,17 @@ protected function validateItems($value, $schema = null, JsonPointer $path = nul
             }
         }
     }
+
+    const OPTIMIZED_ITEM_TYPES = array('string', 'number', 'integer');
+
+    /**
+     * @param \stdClass $schema
+     * @return bool
+     */
+    private function isSupported($schema)
```

Comment:
`isSupported` is a little vague. Can you think of a better function name that better describes what it does? Given that you only call it in one place, another option would be to just remove it and inline the code you have here.